runtime/vm/JFRChunkWriter.cpp

```diff
@@ -815,4 +819,44 @@ VM_JFRChunkWriter::writeInitialSystemPropertyEvents(J9JavaVM *vm)
 	}
 }
 
+
+void
+VM_JFRChunkWriter::writeNativeLibraryEvents()
+{
+#if defined(LINUX)
+	auto dlIterateCallback = [](struct dl_phdr_info *info, size_t size, void *data) -> int {
+		if (NULL == info->dlpi_name || info->dlpi_name[0] == '\0') {
+			return 0;
+		}
+
+		UDATA baseAddress = 0;
+		UDATA topAddress = 0;
+
+		for (int i = 0; i < info->dlpi_phnum; i++) {
+			const ElfW(Phdr) *phdr = info->dlpi_phdr + i;
+			if (PT_LOAD == phdr->p_type) {
+				UDATA phdrBase = info->dlpi_addr + phdr->p_vaddr;
+				UDATA phdrTop = phdrBase + phdr->p_memsz;
+
+				if (0 == baseAddress || phdrBase < baseAddress) {
```

Comment:
requested edit:
`if ((0 == baseAddress) || (phdrBase < baseAddress)) {`

& same for the condition immediately below this one.